iuf

```diff
@@ -566,6 +544,61 @@ def main():
 
     atexit.register(log_state_files, config)
 
+    if config.args["write_input_file"]:
+        configfile.write(config.args)
+
+
+    def process_ctrl_c(sig, frameg):
+        """Override a ctrl-c interrupt."""
+        global already_answered, answer_text
+
+        if already_answered and answer_text:
+            print(answer_text)
+            return
+
+        proccessing_interrupt = True
+        activity = config.args.get("activity", None)
+        script_name = sys.argv[0]
+        print("Would you like to abort this run?")
+        print("    Enter Y, y, or yes to abort after the current stage")
+        print("    Enter F, f, or force to abort immediately.")
+        print("    Enter B, b, or background to exit this script and leave argo running in the background.")
+        print("")
+        print("    Enter anything else to continue without interruption.")
+        print("    Note all logging will be suspended when backgrounded.")
+        literal_answer = get_answer()
+        answer = literal_answer.lower()
+
+        if answer in ["y", "yes"]:
+            answer_text = "Aborting after the current stage completes."
+            if not already_answered:
+                print(answer_text)
+                already_answered = True
+                process_abort(config)
+
+        elif answer in ["b", "background"]:
+            answer_text = "Preparing to exit cli"
+            print(answer_text)
+            if not already_answered:
+                already_answered = True
+                config.activity.abort_activity(config, background_only=True)
+                print(f"Use `{script_name} -a {activity} resume` to re-connect")
```

Comment:
Is that all the arguments needed, or do they need to specify any of the other arguments they originally used before they did a suspend?

Reply:
I believe that's all they need.